src/elements/content-sidebar/BoxAISidebar.tsx

```diff
@@ -1,54 +1,86 @@
 import { mark } from '../../utils/performance';
-import { useFeatureEnabled } from '../common/feature-checking';
+import { BoxAISidebarContext } from './BoxAISidebarContainer';
 
 import messages from '../common/messages';
 import sidebarMessages from './messages';
+
 import './BoxAISidebar.scss';
 
+
 const MARK_NAME_JS_READY: string = `${ORIGIN_BOXAI_SIDEBAR}_${EVENT_JS_READY}`;
 
 mark(MARK_NAME_JS_READY);
+    
+function BoxAISidebar(props: ApiWrapperProps) {
+    const { 
+        createSession, 
+        encodedSession, 
+        onClearAction, 
+        getAIStudioAgents, 
+        isAIStudioAgentSelectorEnabled, 
+        onSelectAgent, 
+        questions, 
+        sendQuestion, 
+        stopQuestion, 
+        ...rest 
+    } = props;
+    const { formatMessage } = useIntl();
+    const { cache, contentName, elementId, setCacheValue, userInfo } = React.useContext(BoxAISidebarContext);
+    const { questions: cacheQuestions } = cache;
 
-export interface BoxAISidebarProps {
-    onClearClick: () => void;
-    agents: AgentType[];
-    selectedAgent: AgentType | null;
-}
+    if (!cache[encodedSession] && encodedSession) {
+        setCacheValue('encodedSession', encodedSession);
+    }
 
-function BoxAISidebar({ agents = [], onClearClick, selectedAgent}: BoxAISidebarProps) {
-    const { formatMessage } = useIntl();
-    const isAgentSelectorEnabled = useFeatureEnabled('boxai.agentSelector.enabled');
+    if (!cache[questions] && questions) {
+        setCacheValue('questions', questions);
+    }
+
+    React.useEffect(() => {
+        if (!encodedSession && createSession) {
+            createSession();
+        }
+
+        if (cacheQuestions.length > 0 && cacheQuestions[cacheQuestions.length-1].isCompleted === false) {
+            // if we have cache with question that is not completed resend it to trigger an API
+            sendQuestion({prompt: cacheQuestions[cacheQuestions.length-1].prompt});
+        }
+
+        return () => {
+            // stop API request on unmount (e.g. during switching to another tab)
+            stopQuestion();
+        }
+    }, []);
 
     const renderBoxAISidebarTitle = () => {
         return (
             <div className="bcs-BoxAISidebar-title-part">
                 <Text as="h3" className="bcs-title">
                     {formatMessage(messages.sidebarBoxAITitle)}
                 </Text>
-                {isAgentSelectorEnabled &&
-                        <BoxAiAgentSelector
-                            agents={agents}
-                            onErrorAction={() => null}
-                            requestState={REQUEST_STATE.SUCCESS}
-                            selectedAgent={selectedAgent}
-                            triggerChipClassName="sidebar-chip" />
+                {isAIStudioAgentSelectorEnabled &&
+                    <BoxAiAgentSelectorWithApi
+                        fetcher={getAIStudioAgents}
+                        onSelectAgent={onSelectAgent}
```

Comment:
while these are the only strictly necessary props, there are others that should be included here: https://git.dev.box.net/Box/frontend-mono/blob/aca02607f44492822a372efe33a60a97ace03822/packages/group-shared-features/box-ai-content-answers/src/lib/components/modal/modal.tsx#L165

Reply:
Added 👍